### ISSUE TITLE
Single quote escaping with `singleQuotes: true`

### ISSUE THREAD

**Issue by Gottwik** [yeoman/stringify-object#40]
I believe there is a bug

```js
var stringify_object = require("stringify-object")

var dummy_object = {
    a: "aa\'aa"
}

console.log(stringify_object(dummy_object, { singleQuotes: true }))
```

notice that the dummy_object is **perfectly ok**, because the single quote is inside double quotes

returns this:
```js
{
    a: 'aa\\'aa'
}
```

which is **malformed** because of the unescaped single quote

**Comment by sindresorhus** [comment]
This is what I get, which seems correct:

```js
console.log(stringifyObject({a: "aa\'aa"}, {singleQuotes: true}));
```
```
{
	a: 'aa\'aa'
}
```

**Comment by Gottwik** [comment]
Your output is fine as the backslash escapes the single quote. I am getting two backslashes, see here:

https://runkit.com/5889faa0646a3a00145b6a53/5889fc1038228e0015882c73

**Comment by Gottwik** [comment]
Interestingly the [first test is expecting](https://github.com/yeoman/stringify-object/blob/master/test.js#L50) `{ foo: 'a \' b \' c \\\' d' }` to be processed into
```js
{
	foo: 'a \' b \' c \\' d'
}
```
Which is not a valid javascript syntax. Am I not getting something?

In any case, I believe I fixed this in #41
